python/pyspark/sql/functions.py

```diff
@@ -1862,9 +1862,33 @@ def udf(f, returnType=StringType()):
     >>> slen = udf(lambda s: len(s), IntegerType())
     >>> df.select(slen(df.name).alias('slen')).collect()
     [Row(slen=5), Row(slen=3)]
+    >>> @udf
+    ... def to_upper(s):
+    ...     if s is not None:
+    ...         return s.upper()
+    ...
+    >>> @udf(returnType=IntegerType())
+    ... def add_one(x):
+    ...     if x is not None:
+    ...         return x + 1
+    ...
     """
     return UserDefinedFunction(f, returnType)
 
+
+@functools.wraps(_udf)
+def udf(f=None, returnType=StringType()):
```

Comment:
Should we keep the since annotation? Also maybe explain what these params are in the docstring (which can be based on the docstirng you currently have for _udf but updated).